### ISSUE TITLE
Suggestion: Add option to allow players to request a contested roll

### ISSUE THREAD

**Issue by MaxEhrhart** [ironmonk88/monks-tokenbar#91]
Hello, I have a suggestion that I think would be good to add and simple to implement.
 
Commonly, the player has features or skills that request contested roll,
for example, Inquisitive Rogue with Insightful Fighting that contests the player Insight vs creature deception.
There are a lot more similar cases that players would ask for a contested roll.

It would be great to have an option in settings to enable the button of a contested roll to the players so the gm does not need to be asked to do it every time the player uses that skill.
![image](https://user-images.githubusercontent.com/22263628/120113751-53107a80-c152-11eb-8ee1-28513b962bf3.png)

And another thing, it would be great to save templates of contested rolls, using the same example as above, the inquisitive rogue uses insightful fighting frequently, so an option to save a template would be great too.

Thx.

**Comment by GabrielDeschain** [comment]
I liked the template and macro suggestions! It would be great to have, for example, initiative, perception  and this stuff request roll templates,   in the macro bar. So it would be possible to make these common requests that we do all the time with 1 click

**Comment by thatlonelybugbear** [comment]
I agree with the concept of players (or at least trusted players and above) being able to initiate contested rolls and save rolls as well. 

For the example of MaxEhrhart I would actually prefer to use a macro as a Macro on Use: ItemMacro (midi-qol and itemmacro modules).
```javascript
let A = canvas.tokens.controlled[0];
let B = game.user.targets[0];
game.MonksTokenBar.requestContestedRoll({token:A,request:'skill:ins'},{token:B,request:'skill:dec'},{silent:true})
```

That right now is not possible because there is a check taking place in the API that needs the user to be a GM.
In the monks-tokenbar-api.js and lines 68 - 70 (the sane goes for requesting saving throws).

```javascript
static requestContestedRoll(request0, request1, options = {}) {
        if (!game.user.isGM)
        return;
```

If you just delete this entry the problem is somewhat mitigated as a GM could make the items + macros for the players to use without giving permissions to players to fully be able to initiate Contested Rolls and Saving throws, if that for any reason would be like opening a can of worms :D

**Comment by FFZendikar** [comment]
> I agree with the concept of players (or at least trusted players and above) being able to initiate contested rolls and save rolls as well.
> 
> For the example of MaxEhrhart I would actually prefer to use a macro as a Macro on Use: ItemMacro (midi-qol and itemmacro modules).
> 
> ```js
> let A = canvas.tokens.controlled[0];
> let B = game.user.targets[0];
> game.MonksTokenBar.requestContestedRoll({token:A,request:'skill:ins'},{token:B,request:'skill:dec'},{silent:true})
> ```
> 
> That right now is not possible because there is a check taking place in the API that needs the user to be a GM. In the monks-tokenbar-api.js and lines 68 - 70 (the same goes for requesting saving throws).
> 
> ```js
>         if (!game.user.isGM)
>         return;
> ```
> 
> If you just get rid of this if check (I just comment the lines out in the .js file) the problem is somewhat mitigated. A GM could make the items + macros for the players to use without giving permission to players to initiate Contested Rolls and Saving throws from the tokenbar, in case that this for any reason would be like opening a can of worms :D

I know this issue is a bit dated, but I was wondering about your macro set up if you would be so kind. using it myself, my only issue is that the game.user.targets[0] doesn't adhere to the request that was set and gets defaulted to a strength ability check. Any guesses why, or is it not a problem for you?

**Comment by thatlonelybugbear** [comment]
@FFZendikar 
Hey, if you want, ping me in the main Foundry discord channel, or Monk's discord channel, or DM directly `@thatlonelybugbear `
so as not to forget and I will get back to you when I can.
